src/common/Compiler.h

```diff
@@ -106,7 +106,11 @@ int CountTrailingZeroes(unsigned long long x);
 #ifdef __SANITIZE_ADDRESS__ // Detects GCC asan
 #   define USING_ADDRESS_SANITIZER
 #elif defined(__has_feature)
-#   if __has_feature(address_sanitizer) // Detects Clang asan
+#   if __has_feature(address_sanitizer) // Detects Clang AddressSanitizer
+#       define USING_ADDRESS_SANITIZER
+#   elif __has_feature(memory_sanitizer) // Detects Clang MemorySanitizer
+#       define USING_ADDRESS_SANITIZER
```

Comment:
nit: does this need to be renamed?

Reply:
For now it works, duplicating all those definitions would be very noisy.

The primary usage is to avoid redefining `new` and `delete`, so it is to always do the same on that side and having 4 defininitions to prevent redefining `new` and `delete` looks overkill.

The secondary usage is for to add `ATTRIBUTE_NO_SANITIZE_ADDRESS` to some function, I have not tested if such function would work on other sanitizers without tht.